tests/functional/parser/test_apigw.py

```diff
@@ -100,3 +103,20 @@ def test_apigw_event():
     assert request_context.operationName is None
     assert identity.apiKey is None
     assert identity.apiKeyId is None
+
+
+def test_apigw_event_with_invalid_websocket_request():
+    # GIVEN an event with an eventType != MESSAGE and has  a messageId
```

Comment:
@risenberg-cyberark - this is how i am validating this?

Reply:
you should provide a full valid event with just event type being not aligned otherwise it would fail way before it gets here

Reply:
@risenberg-cyberark  - i thought of doing that, but this is just a minimal example. In the assert i do check if the specific message is retuned. Which it is after the changes

Reply:
I dont know how it affects the order the validation since it's not documented in pydantic. The root validator is supposed to run after the entire item has passed validation, which with your input, it doesnt as it's a partial event.
that's why in docs you can have root_validator(pre=True)

Reply:
anyway, it's not the expected behavior of the validator here, i'd suggest we either move it back and fix it or remove it completly if you dont think it's valuable enough

Reply:
I will tweak it and see if it makes sense. Otherwise we can remove it, as it is definitely is not working as is. Interesting placing with it none the less.

Reply:
@risenberg-cyberark  i have asserted that there is only one error now.